src/careamics/prediction_utils/lvae_prediction.py

```diff
@@ -0,0 +1,91 @@
+"""Module containing pytorch implementations for obtaining predictions from an LVAE."""
+
+from typing import Optional
+
+import torch
+
+from careamics.models import LVAE
+
+# TODO: convert these functions to lightning module `predict_step`
+#   -> mmse_count will have to be an instance attribute?
+
+
+def lvae_predict_single_sample(
+    model: LVAE, input: torch.Tensor
+) -> tuple[torch.Tensor, Optional[torch.Tensor]]:
+    """
+    Generate a single sample prediction from an LVAE model, for a given input.
+
+    Parameters
+    ----------
+    model : LVAE
+        Trained LVAE model.
+    input : torch.tensor
+        Input to generate prediction for. Expected shape is (S, C, Y, X).
+
+    Returns
+    -------
+    tuple of (torch.tensor, optional torch.tensor)
+        The first element is the sample prediction, and the second element is the
+        log-variance. The log-variance will be None if `model.predict_logvar is None`.
+    """
+    model.eval()  # Not in original predict code: effects batch_norm and dropout layers
+    with torch.no_grad():
+        output: torch.Tensor
+        output, _ = model(input)  # 2nd item is top-down data dict
+
+    # TODO: what if this is likelihood_NM (Not implemented at the moment)
+    # not sure how this will be fully implemented
+    #   -> potentially log_var is always None ?
+    likelihood_obj = model.likelihood_gm
+
+    sample_prediction: torch.Tensor
+    log_var: Optional[torch.Tensor]
```

Comment:
Is this declaration made for mypy? I think it should be fine since the return types are specified in get_mean_lv

Reply:
Ah I think this was left over from when I wasn't using the `get_mean_lv` method, can remove.